gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/HeaderFlagPopoverPanel.java

```diff
@@ -42,18 +45,15 @@ public String getMessageKey() {
 
 	Flag flag;
 	Long assignmentId;
+	GbRole role;
 
-	public HeaderFlagPopoverPanel(final String id, final Flag flag, final Long assignmentId) {
-		super(id);
-
-		this.flag = flag;
-		this.assignmentId = assignmentId;
-	}
-
-	public HeaderFlagPopoverPanel(final String id, final Flag flag) {
-		super(id);
+	public HeaderFlagPopoverPanel(final String id, final IModel<Map<String, Object>> model) {
```

Comment:
I've rejigged `HeaderFlagPopoverPanel` to take an `IModel<Map>` which has the role available on it.. saving another service for each header flag when deciding whether to show the Edit Settings link.